commands.go

```diff
@@ -152,6 +152,13 @@ type Cmdable interface {
 	SRem(key string, members ...interface{}) *IntCmd
 	SUnion(keys ...string) *StringSliceCmd
 	SUnionStore(destination string, keys ...string) *IntCmd
+	XAdd(key string, id string, els ...interface{}) *StringCmd
+	XAddMaxLen(key string, id string, maxLen int64, approximate bool, els ...interface{}) *StringCmd
+	XLen(key string) *IntCmd
+	XRange(key string, start, stop string) *XMessageSliceCmd
+	XRangeN(key string, start, stop string, count int64) *XMessageSliceCmd
+	XRead(block int64, streams []XStreamCheckpoint) *XStreamSliceCmd
+	XReadN(block, count int64, streams []XStreamCheckpoint) *XStreamSliceCmd
```

Comment:
We almost certainly want `key string` to be the first argument so we can retrieve for in Redis Cluster in order to shard data - see https://github.com/go-redis/redis/blob/master/cluster.go#L543-L545. Having key in `XStreamCheckpoint` will complicate that code a lot for little/no reason.

Reply:
Solid point -- strangely though, it seems like the `XREAD` command is intended to allow reads from multiple streams at once, e.g. 
```
> XREAD BLOCK 5000 STREAMS mystream otherstream $ 1506935385635.0
```
Will the cluster support in this library allow this kind of multiplexing?

Reply:
It expects that if there are multiple keys then all keys must be on the same shard.